apps/api/v2/src/modules/auth/guards/organizations/is-admin-access.guard.ts

```diff
@@ -0,0 +1,61 @@
+import { OrganizationsRepository } from "@/modules/organizations/organizations.repository";
+import { RedisService } from "@/modules/redis/redis.service";
+import { Injectable, CanActivate, ExecutionContext, ForbiddenException } from "@nestjs/common";
+import { Request } from "express";
+
+import { Team } from "@calcom/prisma/client";
+
+type CachedData = {
+  org?: Team;
+  canAccess?: boolean;
+};
+
+@Injectable()
+export class IsAdminAccessGuard implements CanActivate {
+  constructor(
+    private organizationsRepository: OrganizationsRepository,
+    private readonly redisService: RedisService
+  ) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    let canAccess = false;
+    const request = context.switchToHttp().getRequest<Request & { organization: Team }>();
+    const organizationId: string = request.params.orgId;
+
+    if (!organizationId) {
+      throw new ForbiddenException("No organization id found in request params.");
+    }
+
+    const REDIS_CACHE_KEY = `apiv2:org:${organizationId}:guard:isAdminAccess`;
+    const cachedData = await this.redisService.redis.get(REDIS_CACHE_KEY);
+
+    if (cachedData) {
+      const { org: cachedOrg, canAccess: cachedCanAccess } = JSON.parse(cachedData) as CachedData;
+      if (cachedOrg?.id === Number(organizationId) && cachedCanAccess !== undefined) {
+        request.organization = cachedOrg;
+        return cachedCanAccess;
+      }
+    }
+
+    const org = await this.organizationsRepository.findById(Number(organizationId));
+
+    if (org?.isOrganization) {
```

Comment:
Platform customers' organizations also have `isOrganization === true`, so either:
1. We should ignore this check if `org.isPlatform` is true
2. When platform team is created via create teams handler we set `adminAPIAccessIsEnable` as true for the platform org.

Reply:
Because right now as is I think all platform calls to ESSENTIALS+ plans invoking orgs endpoints would fail.